### DISCUSSION TITLE
how to get selected item's id instead of whole object in p-autocomplete

### DISCUSSION THREAD

**Discussion by TechMARK-1978** [primefaces/community#126] [PrimeNG] [answered]
<p-autoComplete formControlName="city" [suggestions]="filteredCities"
                        (completeMethod)="filterCity($event)" dropdown="true" field="nam" 
                        tabIndex="8"></p-autoComplete>

when i console the form value it return like this

{
    "id": "0",
    "cmpCode": "KMES",
    "nam": "KMES SOCIETY",
    "printName": "KMES SOCIETY",
    "societyId": "",
    "societyD": "",
    "registrationNo": "123123",
    "addressLine1": "1232132",
    "addressLine2": "123123",
    "pincode": "421302",
    "city": {
        "id": 1,
        "code": "BWD",
        "nam": "BHIWANDI",
        "distID": 14,
        "distD": ""
    },
    "cityD": "",
}

and i want like below
{
    "id": "0",
    "cmpCode": "KMES",
    "nam": "KMES SOCIETY",
    "printName": "KMES SOCIETY",
    "societyId": "",
    "societyD": "",
    "registrationNo": "123123",
    "addressLine1": "1232132",
    "addressLine2": "123123",
    "pincode": "421302",
    "city":1,
    "cityD": "",
    "dist": "",
    "distD": ""
}

this is my code if i select an item than i want to get id instead of whole object plz suggest me a solution 
Thanks

**Comment by cetincakiroglu** [comment, marked as answer]
Unfortunately, you cannot get just the selected ID, the component emits the selected value as an object. In the case above, you need to either update the structure of your data or write a logic that takes the emitted value from the component, and modifies and returns the value you wanted. The component's responsibility is just emitting a selected value, further logic is up to you with that value.

**Reply by TechMARK-1978** [reply]
Thank you for reply i will do as u suggested
